pkg/backup/backupdest/backup_destination.go

```diff
@@ -224,8 +233,39 @@ func ResolveDest(
 	}
 	prevBackupURIs = append([]string{plannedBackupDefaultURI}, prevBackupURIs...)
 
+	baseEncryptionOptions, err := backupencryption.GetEncryptionFromBase(
+		ctx, user, execCfg.DistSQLSrv.ExternalStorageFromURI, prevBackupURIs[0],
+		encryption, kmsEnv,
+	)
+	if err != nil {
+		return ResolvedDestination{}, err
+	}
+
+	if startTime.IsEmpty() {
+		// TODO (kev-cao): Once we have completed the backup directory index work, we
+		// can remove the need to read an entire backup manifest just to fetch the
+		// start time. We can instead read the metadata protobuf.
+		mem := execCfg.RootMemoryMonitor.MakeBoundAccount()
+		defer mem.Close(ctx)
+		precedingBackupManifest, size, err := backupinfo.ReadBackupManifestFromURI(
+			ctx, &mem, prevBackupURIs[len(prevBackupURIs)-1], user,
+			execCfg.DistSQLSrv.ExternalStorageFromURI, baseEncryptionOptions, kmsEnv,
+		)
+		if err != nil {
+			return ResolvedDestination{}, err
+		}
+		if err := mem.Grow(ctx, size); err != nil {
+			return ResolvedDestination{}, err
+		}
+		defer mem.Shrink(ctx, size)
+		startTime = precedingBackupManifest.EndTime
+		if startTime.IsEmpty() {
+			return ResolvedDestination{}, errors.Errorf("empty end time in prior backup manifest")
+		}
+	}
 	// Within the chosenSuffix dir, differentiate incremental backups with partName.
-	partName := endTime.GoTime().Format(backupbase.DateBasedIncFolderName)
```

Comment:
we need to add a 25.2 version gate here: in a mixed version cluster, we would not want a 25.2 node to begin the backup (writing the fancy new suffix), have that node die and have a helpless 25.1 node pick up the job and have no idea how to resolve its destination.

Reply:
version gate looks good! let me know if you want to hop on a call to chat about any general questions about versioning in crdb